### PR TITLE
Fix license file

### PR DESCRIPTION
**Pull request by jtattermusch**
See grpc/grpc-go#1309

New LICENSE file is copied from http://www.apache.org/licenses/LICENSE-2.0.txt

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -171,7 +176,18 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (mehrdada) [inline LICENSE]: Should the template be specialized here or copied verbatim with the placeholders?

File: tools/distrib/check_copyright.py [hunk 2]
```diff
@@ -124,7 +123,7 @@ def save(name, text):
   with open(name, 'w') as f:
     f.write(text)
 
-assert(re.search(RE_LICENSE['LICENSE'], load('LICENSE')))
+
```
Comment (mehrdada) [inline tools/distrib/check_copyright.py]: nit: remove blank line

File: LICENSE (hunk 1 above)
Comment (jtattermusch) [inline LICENSE]: Other projects (like kubernetes) also have the verbatim text from http://www.apache.org/licenses/LICENSE-2.0.txt with no modifications.

Comment (mehrdada) [inline LICENSE]: Then our Go repo is being inconsistent here 🤔

**Comment by jtattermusch** [comment]
Sanity passing, merging.

File: LICENSE (hunk 1 above)
Comment (jtattermusch) [inline LICENSE]: @mehrdada  you're welcome to fix that ;-)

**Comment by mehrdada** [comment]
Done already! https://github.com/grpc/grpc-go/pull/1329

**Pull request merged**